internal/handlers/common/update.go

```diff
@@ -899,10 +899,69 @@ func newUpdateError(code commonerrors.ErrorCode, msg, command string) error {
 	return commonerrors.NewWriteErrorMsg(code, msg)
 }
 
+// newPathValidator creates a root of pathValidator.
+// It shouldn't be used as a node of other pathValidator.
+func newPathValidator() *pathValidator {
+	return &pathValidator{
+		paths: map[string]*pathValidator{},
+		root:  true,
+	}
+}
+
+// pathValidator stores validated paths and checks for any conflicts
+// between them. The conflict is when one path points to the part of
+// another path.
+//
+// Examples of conflicts:
+// `foo.bar, foo.bar`
+// `foo.bar, foo.bar.x`
+//
+// This is not a conflict:
+// `foo.bar.x, foo.bar.y`.
+type pathValidator struct {
+	paths map[string]*pathValidator
+	last  bool
+	root  bool
+}
+
+// validate takes a slice of path keys and checks if the path
+// conflicts with previously validated path. If it's not, then
+// the path is saved in pathValidator.
+func (tree *pathValidator) validate(keys []string) bool {
+	if len(keys) == 0 && !tree.root {
+		// if all keys were proceed correctly - path is not conflicting
+		tree.last = true
+		return true
+	}
+
+	key, keys := keys[0], keys[1:]
+
+	if node, ok := tree.paths[key]; ok {
+		// if path exists in validator and it's the last element, conflict is detected
+		if node.last {
+			return false
+		}
+
+		// if there's no more keys in current path,
+		// but child path exists, conflict is detected
```

Comment:
I think we should use similar terminology as the code. Does _child path exists_ mean node exists right?